### PR TITLE
Update k8s libs to 1.27.2

### LINKED ISSUE #6381: Meta Issue for K8S 1.27
## Meta Issue for k8s 1.27 bump
In order to bump Operator SDK to support Kubernetes 1.27 there are a few dependencies we rely on making the bump first. 

This issue is meant to help track all dependencies and the status of their bumps.

### Order of dependency bumps
**Step 1: upgrade controller-runtime, controller-tools, helm, and operator-framework/operator-manifest-tools(can be done in parallel)**
- [x] sigs.k8s.io/controller-runtime 
- [x] sigs.k8s.io/controller-tools 
- [x] operator-framework/operator-manifest-tools

**Step 2: upgrade operator-framework/api and sigs.k8s.io/kubebuilder-declarative-pattern (can be done immediately after controller-runtime)**
- [x] operator-framework/api
- [x] sigs.k8s.io/kubebuilder-declarative-pattern 

**Step 3.1: upgrade operator-framework/operator-registry and operator-framework/operator-lib (can be done immediately after operator-framework/api)**
- [x] operator-framework/operator-registry https://github.com/operator-framework/operator-registry/pull/1120
- [x] operator-framework/operator-lib

**Step 3.2: upgrade kubebuilder (blocked until kubebuilder-declarative-pattern is bumped)**
- [x] sigs.k8s.io/kubebuilder/v3 

**Step 4: upgrade operator-framework/helm-operator-plugins and operator-framework/java-operator-plugins**
- [x] operator-framework/helm-operator-plugins https://github.com/operator-framework/helm-operator-plugins/pull/216
- [x] operator-framework/java-operator-plugins https://github.com/operator-framework/java-operator-plugins/pull/119

**Step 5:  upgrade operator-framework/operator-sdk dependencies**
- [x] Bump Ginkgo/v2 and Kubebuilder 1.27
- [ ] Bump SDK to use k8s 1.27
- [ ] Bump envtest to 1.27

### PR DESCRIPTION
**Pull request by Neo2308**
**Description of the change:**
* Updated k8s libs to 1.27.2
* Bump Ginkgo/v2 and Kubebuilder versions
* kustomize/v2-alpha->kustomize/v2
* Fixes in helm controller

**Motivation for the change:**
Closes #6381

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [ ] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

## Pending
1. Mitigate controller runtime breaking changes
    - [ ] https://github.com/kubernetes-sigs/controller-runtime/pull/2150
    - [ ] https://github.com/kubernetes-sigs/controller-runtime/pull/2120
    - [ ] Anything else?

2. Others:
    - [ ] Generate manifest files
    - [ ] Envtest bump
    - [ ] Address TODO: go v4 alpha -> stable
    - [ ] Checklist items
    - [ ] Anything else?

### CODE REVIEW

**Comment by Neo2308** [comment]
Changes are not fully complete. Need help with the rest of the PR.

File: internal/cmd/operator-sdk/cli/cli.go [hunk 1]
```diff
@@ -77,47 +77,59 @@ func Run() error {
 // This CLI can run kubebuilder commands and certain SDK specific commands that are aligned for
 // the kubebuilder project layout
 func GetPluginsCLIAndRoot() (*cli.CLI, *cobra.Command) {
-	gov2Bundle, _ := plugin.NewBundle(golang.DefaultNameQualifier, golangv2.Plugin{}.Version(),
-		golangv2.Plugin{},
-		envtestv1.Plugin{},
-		manifestsv2.Plugin{},
-		scorecardv2.Plugin{},
+	gov2Bundle, _ := plugin.NewBundleWithOptions(plugin.WithName(golang.DefaultNameQualifier), plugin.WithVersion(golangv2.Plugin{}.Version()),
```
Comment (Neo2308) [inline internal/cmd/operator-sdk/cli/cli.go:80]: Ref: https://github.com/kubernetes-sigs/kubebuilder/pull/3360

**Comment by varshaprasad96** [comment]
@Neo2308 Here is a WIP PR to bump these dependencies: https://github.com/operator-framework/operator-sdk/pull/6514/files. It was blocked by a Controller Runtime PR related to logging handler that affects ansible operator. I had been working on a workaround, but got preempted by other tasks. I am planning to take some time today/tomorrow to get back to it and take it to the finish line. In case, I'm not able, I can ping you here, and you can work on that PR from where I left, I can help in making it to the finish line. Does that sound good ?

**Comment by Neo2308** [comment]
Not sure how I missed your PR. Yours is much further along. It's just missing a couple of small fixes it seems like. Let me know if you need any help  🙌

**Comment by Neo2308** [comment]
Closing this PR since its been covered in https://github.com/operator-framework/operator-sdk/pull/6613 and available in https://github.com/operator-framework/operator-sdk/releases/tag/v1.33.0

**Pull request closed without merging**